keras/engine/topology.py

```diff
@@ -3042,26 +3042,104 @@ def preprocess_weights_for_loading(layer, weights,
             if layer.__class__.__name__ == 'ConvLSTM2D':
                 weights[1] = np.transpose(weights[1], (3, 2, 0, 1))
 
-    # convert the weights of CuDNNLSTM so that they could be loaded into LSTM
-    if layer.__class__.__name__ == 'LSTM' and len(weights) == 3:
+    weights = _convert_rnn_weights(layer, weights)
+
+    return weights
+
+
+def _convert_rnn_weights(layer, weights):
+    """Converts weights for RNN layers between native and CuDNN format.
```

Comment:
Please format the docstring like the others, with an `# Arguments` and `# Returns` section